gio/src/list_store.rs

```diff
@@ -43,20 +44,28 @@ impl ListStore {
         }
     }
 
+    /// Dumbest possible example just to test this. Will remove.
+    /// ```
+    /// use glib::prelude::*;
+    /// use gio::ListStore;
+    /// use gio::prelude::*;
+    /// let item1 = ListStore::new(ListStore::static_type());
+    /// let item2 = ListStore::new(ListStore::static_type());
+    /// let list = ListStore::new(ListStore::static_type());
+    /// list.splice(0, 0, &[item1, item2.clone()]);
+    /// assert!(list.item(1) == Some(item2.upcast()))
+    /// ```
     #[doc(alias = "g_list_store_splice")]
     pub fn splice(&self, position: u32, n_removals: u32, additions: &[impl IsA<glib::Object>]) {
```

Comment:
It might be worth adding a `fn extend_from_slice(&self, &[impl IsA<Object>])` convenience function as the position/removals usage is not too obvious and requires a few seconds of thinking.

Reply:
It's now covered by `extend`. See latest commit.
These now work:
```
list.extend([obj, obj]);
list.extend(&[obj, obj]);
list.extend([&obj, &obj]);
```

Reply:
API-wise yes, but just like `Vec` has an `extend_from_slice()` for efficiency reasons, it also makes sense here. You wouldn't need to create a temporary `Vec` for that but could directly call into `splice()`.